spec/controllers/equipment_models_controller_spec.rb

```diff
@@ -1,421 +1,387 @@
-        expect(assigns(:equipment_models).include?(mod_other_cat_inactive))
-          .to be_truthy
-        expect(assigns(:equipment_models).size).to eq(3)
-      end
-    end
+
+  def mock_category(**attrs)
+    cat = instance_spy('Category', **attrs)
+    allow(Category).to receive(:find).and_return(cat)
+    cat
   end
-end
 
-describe EquipmentModelsController, type: :controller do
-  before(:each) { mock_app_config }
-  let!(:model) { FactoryGirl.create(:equipment_model) }
+  def mock_models_with_category(cat: nil, models: nil, count: 1)
+    cat ||= mock_category
+    models ||= Array.new(count) { mock_eq_model }
+    models.each { |model| allow(model).to receive(:category).and_return(cat) }
+    allow(cat).to receive(:equipment_models).and_return(models)
+    models
+  end
 
   it_behaves_like 'calendarable', EquipmentModel
 
+  USER_ROLES = [:admin, :user].freeze
+
   describe 'GET index' do
-    context 'with admin user' do
+    shared_examples_for 'GET index success' do |user_role|
       before do
-        sign_in FactoryGirl.create(:admin)
-        get :index
+        mock_user_sign_in(mock_user(user_role))
       end
-      it_behaves_like 'GET index success'
-    end
-    context 'with non-admin user' do
-      before { sign_in FactoryGirl.create(:user) }
-      describe 'should redirect to root' do
+
+      describe 'basic function' do
         before { get :index }
-        it_behaves_like 'GET index success'
+        it_behaves_like 'successful request', :index
+      end
+
+      context 'with @category set' do
+        it 'populates an array of of active category-type equipment models' do
+          cat = mock_category(id: 1)
+          model = mock_models_with_category(cat: cat)
+          expect(model).to receive_message_chain(:includes, :active)
+          get :index, category_id: cat.id
+        end
+      end
+      context 'without @category set' do
+        it 'populates an array of all active equipment models' do
+          models = mock_eq_collection
+          expect(models).to receive_message_chain(:includes, :active)
+          get :index
+        end
+      end
+      context 'with show deleted' do
+        context 'with @category set' do
+          it 'populates an array of category-type equipment models' do
+            cat = mock_category(id: 1)
+            models = mock_models_with_category(cat: cat)
+            expect(models).to receive(:includes)
+            expect(models).not_to receive(:active)
+            get :index, category_id: cat.id, show_deleted: true
+          end
+        end
+        context 'without @category set' do
+          it 'populates an array of all equipment models' do
+            models = mock_eq_collection
+            expect(models).to receive(:includes)
+            expect(models).not_to receive(:active)
+            get :index, show_deleted: true
+          end
+        end
       end
     end
+    USER_ROLES.each { |type| it_behaves_like 'GET index success', type }
```

Comment:
the above tests are, I think, much better than the ones below. Some of the controller methods are too complex to be reasonably tested with this approach, so I left them in a separate example group for now (they'll be converted when the controller is refactored)